eng/common/templates-official/job/source-index-stage1.yml

```diff
@@ -62,7 +62,24 @@ jobs:
     displayName: Process Binlog into indexable sln
 
   - ${{ if and(eq(parameters.runAsPublic, 'false'), ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
-    - script: $(Agent.TempDirectory)/.source-index/tools/UploadIndexStage1 -i .source-index/stage1output -n $(Build.Repository.Name)
-      displayName: Upload stage1 artifacts to source index
-      env:
-        BLOB_CONTAINER_URL: $(source-dot-net-stage1-blob-container-url)
+    - task: AzureCLI@2
+      displayName: Get stage 1 auth token
+      inputs:
+        azureSubscription: 'SourceDotNet Stage1 Publish'
+        addSpnToEnvironment: true
+        scriptType: 'ps'
+        scriptLocation: 'inlineScript'
+        inlineScript: |
+          echo "##vso[task.setvariable variable=ARM_CLIENT_ID]$env:servicePrincipalId"
+          echo "##vso[task.setvariable variable=ARM_ID_TOKEN]$env:idToken"
```

Comment:
Not sure for Github, but Azure has this `;issecret=true` https://learn.microsoft.com/en-us/azure/devops/pipelines/process/set-secret-variables?view=azure-devops&tabs=yaml%2Cbash#set-secret-variable-in-a-script-with-logging-commands

Does it apply here?

`echo "##vso[task.setvariable variable=ARM_ID_TOKEN;issecret=true]$env:idToken"`

Reply:
It's a federated token, which isn't a secret in the traditional sense.  The whole idea behind federated credentials is to eliminate secret management.

Reply:
See https://github.com/dotnet/arcade/pull/14750/files#r1600647304